src/coreclr/src/vm/diagnosticsprotocol.h

```diff
@@ -24,7 +24,7 @@ bool TryParse(uint8_t *&bufferCursor, uint32_t &bufferLen, T &result)
 
     if (bufferLen < sizeof(T))
         return false;
-    result = *(reinterpret_cast<T *>(bufferCursor));
+    memcpy(&result, bufferCursor, sizeof(T));
```

Comment:
Might consider something like:

```C
template <typename T>
static inline T unaligned_reinterpret_cast(const void* ptr)
{
   T tmp; 
   memcpy(&tmp, ptr, sizeof(T));
   return tmp;
}
```

Then just use it like:

```C
result = unaligned_reinterpret_cast<T>(bufferCursor);
```

At least, for fundamental types such as `int` and `float`, this generates pretty good code that does not involve copies.

Reply:
I'm not sure I see how refactoring that code removes the need for a copy.  `memcpy` happens either way.  Are you suggesting replacing the entirety of `TryParse` with the above code?

Reply:
Oh, I mean, it still calls `memcpy()` (which might not even copy, just load it in a register), but there's no additional copy because it's returning `tmp`.